src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -3192,79 +3191,52 @@ private IOperation CallNullableMember(IOperation value, SpecialMember nullableMe
 
             // Avoid creation of default values and FlowCapture for conditional access on a statement level.
             bool isOnStatementLevel = _currentStatement == operation || (_currentStatement == operation.Parent && _currentStatement?.Kind == OperationKind.ExpressionStatement);
-            var frames = ArrayBuilder<EvalStackFrame>.GetInstance();
+            EvalStackFrame? expressionFrame = null;
+            var operations = ArrayBuilder<IOperation>.GetInstance();
 
             if (!isOnStatementLevel)
             {
-                frames.Push(PushStackFrame());
+                expressionFrame = PushStackFrame();
             }
 
             IConditionalAccessOperation currentConditionalAccess = operation;
 
-            while (true)
-            {
-                frames.Push(PushStackFrame());
-
-                if (currentConditionalAccess.WhenNotNull.Kind != OperationKind.ConditionalAccess)
-                {
-                    break;
-                }
-
-                currentConditionalAccess = (IConditionalAccessOperation)currentConditionalAccess.WhenNotNull;
-            }
-
-            var whenNull = new BasicBlockBuilder(BasicBlockKind.Block);
-
             currentConditionalAccess = operation;
             IOperation testExpression;
+            var whenNull = new BasicBlockBuilder(BasicBlockKind.Block);
+            var previousTracker = _currentConditionalAccessTracker;
+            _currentConditionalAccessTracker = new ConditionalAccessOperationTracker(operations, whenNull);
```

Comment:
NIT: consider moving this below the while loop, to make it clearer how the operations are being mutated. It took me a bit to figure out how the stack was actually getting built.

Reply:
I can't. The loop needs this to be set up correctly in case it encounters a NoneOperation.